subsys/flash_map/flash_map.c

```diff
@@ -0,0 +1,270 @@
+/*
```

Comment:
should not be in its own subsystem, this needs to be either part of fs or a new subsystem with generic name, like storage for example.